lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -857,6 +857,10 @@ function dosomething_reportback_mbp_request($entity) {
       'impact_noun' => $entity->noun,
       'image_markup' => $image_markup,
     );
+    $node = node_load($entity->nid);
```

Comment:
You've been told wrong. Drupal will load it from node cache, which makes this operation lightning fast.